hack/tools/tilt-prepare/main.go

```diff
@@ -496,7 +489,58 @@ func prepareDeploymentForDebug(prefix string, objs []unstructured.Unstructured)
 
 			d.Spec.Template.Spec.Containers[j] = container
 		}
+	})
+}
+
+// prepareDeploymentForObservability alters controller deployments for working
+// nicely with prometheus metrics scraping. Specifically, the metrics endpoint is set to
+// listen on all interfaces instead of only localhost, and another port is added to the
+// container to expose the metrics endpoint.
+func prepareDeploymentForObservability(prefix string, objs []unstructured.Unstructured) error {
+	return updateDeployment(prefix, objs, func(d *appsv1.Deployment) {
+		for j, container := range d.Spec.Template.Spec.Containers {
+			args := make([]string, 0, len(container.Args))
+			for _, a := range container.Args {
+				if a == "--metrics-bind-addr=localhost:8080" {
```

Comment:
should this check for --metrics-bind-addr= prefix vs checking for the value?
also, what if the is no --metrics-bind-addr= in the provider config?
what about rewriting this so we drop the default setting in any, then we add the custom one for tilt always?

Reply:
+1 to match on the prefix in any case

I think there's a trade off:
1. If we only replace it if it's there, then we can be 100% sure that the controller implements that flag
2. If we always add it, that might break some Deployment. On the other side this is a forcing function to provide a metrics endpoint and name the flag consistently :)

If we implement 2., it's good enough to just always append the flag. We don't have to drop existing flags as the flag library will use the value of the last flag if there are multiple.

Reply:
> If we always add it, that might break some Deployment. On the other side this is a forcing function to provide a metrics endpoint and name the flag consistently :)

I'm in favor of this, and in fact, in our migration guide, we provided instruction every time things changed from the controller runtime side, so I assume most of the providers are already there, but might be work make a quick check on all the providers in kubernetes-sigs
however, I'm -1 to always append, because this is confusing if you look at the pod. While we are at it, let's fix ti up properly

Reply:
For now I updated the function to replace the argument based on the `--metrics-bind-addr=` prefix.

Reply:
Thanks for the quick check!
I will open issues in the providers using old names or without --metrics in the manifest advocating for consistency.
+1 to the current implementation given the above context